src/Akeneo/Connectivity/Connection/back/Infrastructure/Install/MigrateAudit40Master.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Infrastructure\Install;
+
+use Akeneo\Connectivity\Connection\Application\Audit\Command\UpdateProductEventCountCommand;
+use Akeneo\Connectivity\Connection\Application\Audit\Command\UpdateProductEventCountHandler;
+use Akeneo\Connectivity\Connection\Domain\Audit\Model\HourlyInterval;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @author Romain Monceau <romain@akeneo.com>
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ *
+ * @todo: Remove this class when pulling up 4.0 on master
+ */
+class MigrateAudit40Master
+{
+    /** @var Connection */
+    private $dbalConnection;
+
+    /** @var UpdateProductEventCountHandler */
+    private $updateProductEventCountHandler;
+
+    public function __construct(
+        Connection $dbalConnection,
+        UpdateProductEventCountHandler $updateProductEventCountHandler
+    ) {
+        $this->dbalConnection = $dbalConnection;
+        $this->updateProductEventCountHandler = $updateProductEventCountHandler;
+    }
+
+    public function migrateIfNeeded(): void
+    {
+        if ($this->needsMigration()) {
+            $createNewAuditTableSql = <<<SQL
+CREATE TABLE IF NOT EXISTS akeneo_connectivity_connection_audit_product(
+    connection_code VARCHAR(100) NOT NULL,
+    event_datetime DATETIME NOT NULL,
+    event_count INT NOT NULL,
+    event_type VARCHAR(100) NOT NULL,
+    updated DATETIME NOT NULL,
+    PRIMARY KEY (connection_code, event_datetime, event_type)
```

Comment:
```suggestion
    PRIMARY KEY (event_datetime, connection_code, event_type)
```
Datetime first to optimize the index/query? (If I understood what Benoit explained).